### DISCUSSION TITLE
Triggering actions by other repository webhooks

### DISCUSSION THREAD

**Discussion by wojtekmach** [community/community#26384] [Actions]
<p>Hi, I’d like to ask if there’s a way of triggering actions from other repository webhooks without leaving GitHub platform.</p>
<p>For example, when there’s a new release on elixir-lang/elixir, I’d like to trigger a workflow in my docker-elixir repository, ideally with the event (ReleaseEvent) payload. Ideally this would be totally decoupled and I could subscribe to any repository webhooks, even those I don’t have write access to.</p>
<p>There’s the <code>repository_dispatch</code> trigger for actions, but I believe it only has event_type field, not the whole payload. Also, I believe it’s not possible to trigger that event from a webhook, I was getting authorization failures.</p>
<p>I noticed there’s a preview support for <a>WebSub</a>, it looks like by subscribing to a topic we (understandably so) set up a webhook, but this means I can only subscribe to repos I am allowed to create webhooks on.</p>
<p>This can be solved today by having a workflow in my docker-elixir that runs on schedule and checks whether there’s a new release in elixir-lang/elixir, and if so it buils the image. If I get elixir-lang/elixir maintainers to set up a webhook, that webhook could hit a web app I maintain that in turn would trigger <code>repository_event</code> which again would check whether there are any new releases. While the 2nd option avoids doing unnecessary checks it brings a ton of complexity.</p>
<p>Are there any other ways of solving this?</p>

**Comment by chrispat** [comment, marked as answer]
<p>You can put a client_payload in a repository_dispatch and use that information during your run.  However, you still need a way to send that event which is going to require your own webhook for now.  This sort of automatic dependency building is something we are looking at for the future but the potential scale concerns are significant.</p>

**Comment by bendavies** [comment]
<p><a class="mention" href="/chrispat">@chrispat</a> i cannot see <code>client\_payload</code> documented anywhere:</p>
<p><a href="https://developer.github.com/v3/repos/#create-a-repository-dispatch-event" rel="nofollow noopener">https://developer.github.com/v3/repos/#create-a-repository-dispatch-event</a></p>
<p>can you clarify?</p>
<p>Thanks!</p>

**Comment by chrispat** [comment]
<p>Unfortuately the docs are lagging a bit.  If you simply add something like</p>
<pre><code>client_payload: { greeting: "hello" }
</code></pre>
<p>to the post body for your request it should be available in the github.event context</p>

**Comment by ciiqr** [comment]
<p>For anyone landing here and needing a more specific example:</p>
<p>send event with body</p>
<pre><code>curl -H "Authorization: token :token" \
-H 'Accept: application/vnd.github.everest-preview+json' \
"https://api.github.com/repos/:owner/:repo/dispatches" \
-d '{"event_type": "awesomeness", "client_payload": {"foo": "bar"}}'
</code></pre>
<p>access fields of the payload directly, or simply the entire payload as json</p>
<pre><code>- run: 'echo "field: ${{ github.event.client_payload.foo }}"'
- run: 'echo "payload: ${{ toJson(github.event.client_payload) }}"'
</code></pre>

**Comment by idemianspro** [comment]
<p>Hello,<br>
The following line works for me:<br>
<code>curl -XPOST -u "${{ secrets.PAT_USERNAME}}:${{secrets.PAT_TOKEN}}" -H "Accept: application/vnd.github.everest-preview+json"  -H "Content-Type: application/json" https://api.github.com/repos/xxx/yyyy/dispatches --data '{"event_type": "build_application"}'</code><br>
work only default baranch <a href="https://api.github.com/repos/xxx/yyyy/" rel="nofollow noopener">https://api.github.com/repos/xxx/yyyy/</a><br>
How to use non-default brunch ?</p>

**Comment by woter1832** [comment]
<p><span class="mention">@Github</span>, has there been any movement on this?</p>
<p>As you refuse to provide a way for us to stop automatic updates of your runner, it would go some way to alleviating the pain if you would providing a webhook that we could use to trigger automatic builds of our containers every time you update your runner.</p>
<p>Thanks</p>
